datalad_next/utils/deprecate.py

```diff
@@ -0,0 +1,89 @@
+import inspect
+import warnings
+from functools import wraps
+
+from datalad_next.utils import get_wrapped_class
+
+
+def deprecated(msg, version, parameter=None, parameter_choice=None):
+    """Annotate functions, classes, or function parameters with standardized
+    deprecation warnings.
+
+    Parameters
+    ----------
+    version: str
+      Software version number at which the deprecation was made
+    msg: str
+      Custom message to append to a deprecation warning
+    parameter: str
+      Individual parameter being deprecated (instead of entire function/class)
+    parameter_choice: str
```

Comment:
I am wondering whether `value` would be better than `choice`. This will often be used for choices, but the implementation uses an equality test, so it would work for anything else too.

The type declaration seems to be incomplete, more than `str` is supported. See below.

Reply:
Adjusted and type-annotated